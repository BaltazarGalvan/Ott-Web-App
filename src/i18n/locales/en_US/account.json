{
  "checkout": {
    "applicable_tax": "Applicable tax ({{taxRate}}%)",
    "close": "Close",
    "continue": "Continue",
    "coupon_discount": "Coupon discount",
    "credit_card": "Credit Card",
    "discount_period": "For the first {{period}}",
    "discount_period_plural": "For {{count}} {{period}}",
    "monthly": "Monthly subscription",
    "no_payment_needed": "No payment needed",
    "payment_cancelled": "Payment cancelled",
    "payment_cancelled_message": "You have cancelled the payment",
    "payment_error": "Payment failed",
    "payment_method": "Payment method",
    "payment_method_fee": "Payment method fee",
    "paypal": "PayPal",
    "paypal_instructions": "Clicking 'continue' will bring you to the PayPal site.",
    "redeem_coupon": "Redeem coupon",
    "start_watching": "Start watching ({{countdown}})",
    "total_price": "Total",
    "welcome_description": "Thank you for subscribing to {{siteName}}. Please enjoy all our content.",
    "welcome_title": "Welcome to {{siteName}}",
    "yearly": "Yearly subscription"
  },
  "choose_offer": {
    "all_movies_and_series_of_platform": "All movies and series of {{siteName}}",
    "benefits": {
      "cancel_anytime": "Cancel anytime",
      "first_days_free": "First day free",
      "first_days_free_plural": "First {{count}} days free",
      "first_periods_free": "First {{period}} free",
      "first_periods_free_plural": "First {{count}} {{period}} days free",
      "watch_on_all_devices": "Watch on all devices"
    },
    "continue": "Continue",
    "field_required": "This field is required",
    "monthly": "Monthly",
    "monthly_subscription": "Monthly subscription",
    "offer_not_found": "Offer not found, please try reloading the page",
    "subscription": "Subscription",
    "yearly": "Yearly",
    "yearly_subscription": "Yearly subscription"
  },
  "login": {
    "email": "Email",
    "field_is_not_valid_email": "Please re-enter your email details and try again.",
    "field_required": "This field is required",
    "forgot_password": "Forgot password?",
    "hide_password": "Hide password",
    "password": "Password",
    "sign_in": "Sign in",
    "view_password": "View password",
    "wrong_combination": "Incorrect email/password combination",
    "wrong_email": "Please check your email and try again."
  },
  "periods": {
    "day": "day",
    "month": "month",
    "week": "week",
    "year": "year",
    "day_plural": "days",
    "week_plural": "weeks",
    "month_plural": "months",
    "year_plural": "years"
  },
  "personal_details": {
    "address": "Address line 1",
    "address2": "Address line 2",
    "birth_date": "Date of birth",
    "birth_date_not_valid": "Birth of date is not a valid date",
    "city": "City",
    "company_name": "Company name",
    "continue": "Continue",
    "fist_name": "First name",
    "last_name": "Last name",
    "no_answer": "No answer",
    "phone_number": "Phone number",
    "post_code": "Zip/Postal Code",
    "state": "State/Province/Region",
    "this_field_is_required": "This field is required",
    "title": "Personal details"
  },
  "registration": {
    "already_account": "Already have an account?",
    "continue": "Continue",
    "email": "Email",
    "field_is_not_valid_email": "Please re-enter your email details",
    "field_required": "This field is required",
    "hide_password": "Hide password",
    "invalid_password": "Use a minimum of 6 characters (case sensitive) with at least one number or special character and one capital character",
    "password": "Password",
    "password_strength": "Use a minimum of 6 characters (case sensitive) with at least one number or special character and one capital character",
    "sign_up": "Sign up",
    "user_exists": "There is already a user with this email address",
    "view_password": "View password",
    "email_updates": "Yes, I want to receive {{siteName}} updates by email."
  },
  "reset": {
    "back_login": "Back to login",
    "confirm": "Confirm",
    "email": "Email",
    "email_me": "Email me",
    "forgot_password": "Forgot password",
    "forgot_text": "We will send you an email with instructions on how to reset your password.",
    "hide_password": "Hide password",
    "invalid_link": "Invalid link",
    "invalid_password": "Invalid password",
    "invalid_token": "Invalid link",
    "link_sent": "Password link sent",
    "link_sent_text": "Please check your inbox at {{email}}",
    "no": "No, thanks",
    "not_sure": "Not sure that was the right email address?",
    "password": "Password",
    "password_reset": "Password reset",
    "reset_password": "Edit Password",
    "text": "If you want to edit your password, click 'YES, Reset' to receive password reset instruction on your mail",
    "try_again": "Try again",
    "view_password": "View password",
    "wrong_email": "Wrong email",
    "yes": "Yes, reset",
    "password_reset_text": ""
  }
}
