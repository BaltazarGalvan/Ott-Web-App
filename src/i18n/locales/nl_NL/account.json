{
  "checkout": {
    "applicable_tax": "",
    "close": "",
    "continue": "",
    "coupon_discount": "",
    "credit_card": "",
    "discount_period": "",
    "discount_period_plural": "",
    "monthly": "",
    "no_payment_needed": "",
    "payment_cancelled": "",
    "payment_cancelled_message": "",
    "payment_error": "",
    "payment_method": "",
    "payment_method_fee": "",
    "paypal": "",
    "paypal_instructions": "",
    "redeem_coupon": "",
    "start_watching": "",
    "total_price": "",
    "welcome_description": "",
    "welcome_title": "",
    "yearly": ""
  },
  "choose_offer": {
    "all_movies_and_series_of_platform": "",
    "benefits": {
      "cancel_anytime": "",
      "first_days_free": "",
      "first_days_free_plural": "",
      "first_periods_free": "",
      "first_periods_free_plural": "",
      "watch_on_all_devices": ""
    },
    "continue": "",
    "field_required": "",
    "monthly": "",
    "monthly_subscription": "",
    "offer_not_found": "",
    "subscription": "",
    "yearly": "",
    "yearly_subscription": ""
  },
  "login": {
    "email": "",
    "field_is_not_valid_email": "",
    "field_required": "",
    "forgot_password": "",
    "hide_password": "",
    "password": "",
    "sign_in": "",
    "view_password": "",
    "wrong_combination": "",
    "wrong_email": ""
  },
  "periods": {
    "day": "",
    "month": "",
    "week": "",
    "year": ""
  },
  "personal_details": {
    "address": "",
    "address2": "",
    "birth_date": "",
    "birth_date_not_valid": "",
    "city": "",
    "company_name": "",
    "continue": "",
    "fist_name": "",
    "last_name": "",
    "no_answer": "",
    "phone_number": "",
    "post_code": "",
    "state": "",
    "this_field_is_required": "",
    "title": ""
  },
  "registration": {
    "already_account": "",
    "continue": "",
    "email": "",
    "field_is_not_valid_email": "",
    "field_required": "",
    "hide_password": "",
    "invalid_password": "",
    "password": "",
    "password_strength": "",
    "sign_up": "",
    "user_exists": "",
    "view_password": "",
    "email_updates": ""
  },
  "reset": {
    "back_login": "",
    "confirm": "",
    "email": "",
    "email_me": "",
    "forgot_password": "",
    "forgot_text": "",
    "hide_password": "",
    "invalid_link": "",
    "invalid_password": "",
    "invalid_token": "",
    "link_sent": "",
    "link_sent_text": "",
    "no": "",
    "not_sure": "",
    "password": "",
    "password_reset": "",
    "reset_password": "",
    "text": "",
    "try_again": "",
    "view_password": "",
    "wrong_email": "",
    "yes": "",
    "password_reset_text": ""
  }
}
