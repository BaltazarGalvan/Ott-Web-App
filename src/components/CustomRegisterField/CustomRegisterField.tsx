import { type FC, type ChangeEventHandler, type ReactNode, useMemo } from 'react';
import type { GetRegisterFieldOption } from '@inplayer-org/inplayer.js';

import Checkbox from '#components/Checkbox/Checkbox';
import TextField from '#components/TextField/TextField';
import Radio from '#components/Radio/Radio';
import Dropdown from '#components/Dropdown/Dropdown';
import { ConsentFieldVariants } from '#src/services/inplayer.account.service';
import { countries, usStates } from '#static/json';

type Props = {
  type: ConsentFieldVariants;
  name: string;
  value: string | boolean;
  onChange: ChangeEventHandler<HTMLInputElement>;
} & Partial<{
  label: ReactNode;
  placeholder: string;
  error: boolean;
  helperText: string;
  disabled: boolean;
  required: boolean;
  options: GetRegisterFieldOption;
}>;

export type CustomRegisterFieldCommonProps = Props;

export const CustomRegisterField: FC<Props> = ({ type, value = '', ...props }) => {
  const optionsList = useMemo(() => {
    const optionsObject = (() => {
      switch (type) {
        case ConsentFieldVariants.COUNTRY_SELECT:
          return countries;
        case ConsentFieldVariants.US_STATE_SELECT:
          return usStates;
        default:
          return props.options;
      }
    })();

    if (!optionsObject) {
      return [];
    }

    return Object.entries(optionsObject).map(([value, label]) => ({ value, label }));
  }, [type, props.options]);

  switch (type) {
    case ConsentFieldVariants.CHECKBOX:
      return <Checkbox {...props} checked={value === true} />;
    case ConsentFieldVariants.INPUT:
      return <TextField {...props} value={value as string} />;
    case ConsentFieldVariants.RADIO:
      return <Radio {...props} values={optionsList} value={value as string} header={props.label} />;
    case ConsentFieldVariants.GENERAL_SELECT:
    case ConsentFieldVariants.COUNTRY_SELECT:
    case ConsentFieldVariants.US_STATE_SELECT:
      return <Dropdown {...props} options={optionsList} value={value as string} defaultLabel={props.placeholder} fullWidth />;
  }

  return null;
};

export default CustomRegisterField;
