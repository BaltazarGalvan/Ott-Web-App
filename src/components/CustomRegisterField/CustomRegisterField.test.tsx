import React from 'react';
import { render } from '@testing-library/react';

import CustomRegisterField from './CustomRegisterField';

import { REGISTER_FIELD_VARIANT } from '#src/services/inplayer.account.service';

describe('<CustomRegisterField>', () => {
  test('renders and matches snapshot <Checkbox>', () => {
    const { container } = render(<CustomRegisterField type={REGISTER_FIELD_VARIANT.CHECKBOX} label="label" name="name" value="value" onChange={vi.fn()} />);

    expect(container).toMatchSnapshot();
  });

  test('renders and matches snapshot <TextField>', () => {
    const { container } = render(<CustomRegisterField type={REGISTER_FIELD_VARIANT.INPUT} label="label" name="name" value="value" onChange={vi.fn()} />);

    expect(container).toMatchSnapshot();
  });

  test('renders and matches snapshot <Radio>', () => {
    const { container } = render(<CustomRegisterField type={REGISTER_FIELD_VARIANT.RADIO} label="label" name="name" value="value" onChange={vi.fn()} />);

    expect(container).toMatchSnapshot();
  });

  test('renders and matches snapshot <Dropdown type="select">', () => {
    const { container } = render(
      <CustomRegisterField type={REGISTER_FIELD_VARIANT.GENERIC_SELECT} label="label" name="name" value="value" onChange={vi.fn()} />,
    );

    expect(container).toMatchSnapshot();
  });

  test('renders and matches snapshot <Dropdown type="country">', () => {
    const { container } = render(
      <CustomRegisterField type={REGISTER_FIELD_VARIANT.COUNTRY_SELECT} label="label" name="name" value="value" onChange={vi.fn()} />,
    );

    expect(container).toMatchSnapshot();
  });

  test('renders and matches snapshot <Dropdown type="us_state">', () => {
    const { container } = render(
      <CustomRegisterField type={REGISTER_FIELD_VARIANT.US_STATE_SELECT} label="label" name="name" value="value" onChange={vi.fn()} />,
    );

    expect(container).toMatchSnapshot();
  });

  test('renders and matches snapshot <Dropdown type="datepicker">', () => {
    const { container } = render(<CustomRegisterField type={REGISTER_FIELD_VARIANT.DATE_PICKER} label="label" name="name" value="value" onChange={vi.fn()} />);

    expect(container).toMatchSnapshot();
  });

  test('renders and matches snapshot <Dropdown type="randomstring">', () => {
    // @ts-ignore
    const { container } = render(<CustomRegisterField type="randomstring" label="label" name="name" value="value" onChange={vi.fn()} />);

    expect(container).toMatchSnapshot();
  });
});
