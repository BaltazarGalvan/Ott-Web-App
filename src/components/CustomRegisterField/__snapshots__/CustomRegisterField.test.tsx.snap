// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CustomRegisterField> > renders and matches snapshot <Checkbox> 1`] = `
<div>
  <div
    class="_checkbox_531f07"
  >
    <div
      class="_row_531f07"
    >
      <input
        id="check-box_1235_name"
        name="name"
        type="checkbox"
        value=""
      />
      <label
        for="check-box_1235_name"
      >
        label
      </label>
    </div>
  </div>
</div>
`;

exports[`<CustomRegisterField> > renders and matches snapshot <Dropdown type="country"> 1`] = `
<div>
  <div
    class="_checkbox_531f07"
  >
    <div
      class="_row_531f07"
    >
      <input
        id="check-box_1235_name"
        name="name"
        type="checkbox"
        value=""
      />
      <label
        for="check-box_1235_name"
      >
        label
      </label>
    </div>
  </div>
</div>
`;

exports[`<CustomRegisterField> > renders and matches snapshot <Dropdown type="datepicker"> 1`] = `
<div>
  <div
    class="_checkbox_531f07"
  >
    <div
      class="_row_531f07"
    >
      <input
        id="check-box_1235_name"
        name="name"
        type="checkbox"
        value=""
      />
      <label
        for="check-box_1235_name"
      >
        label
      </label>
    </div>
  </div>
</div>
`;

exports[`<CustomRegisterField> > renders and matches snapshot <Dropdown type="randomstring"> 1`] = `
<div>
  <div
    class="_checkbox_531f07"
  >
    <div
      class="_row_531f07"
    >
      <input
        id="check-box_1235_name"
        name="name"
        type="checkbox"
        value=""
      />
      <label
        for="check-box_1235_name"
      >
        label
      </label>
    </div>
  </div>
</div>
`;

exports[`<CustomRegisterField> > renders and matches snapshot <Dropdown type="select"> 1`] = `
<div>
  <div
    class="_checkbox_531f07"
  >
    <div
      class="_row_531f07"
    >
      <input
        id="check-box_1235_name"
        name="name"
        type="checkbox"
        value=""
      />
      <label
        for="check-box_1235_name"
      >
        label
      </label>
    </div>
  </div>
</div>
`;

exports[`<CustomRegisterField> > renders and matches snapshot <Dropdown type="us_state"> 1`] = `
<div>
  <div
    class="_checkbox_531f07"
  >
    <div
      class="_row_531f07"
    >
      <input
        id="check-box_1235_name"
        name="name"
        type="checkbox"
        value=""
      />
      <label
        for="check-box_1235_name"
      >
        label
      </label>
    </div>
  </div>
</div>
`;

exports[`<CustomRegisterField> > renders and matches snapshot <Radio> 1`] = `
<div>
  <div
    class="_checkbox_531f07"
  >
    <div
      class="_row_531f07"
    >
      <input
        id="check-box_1235_name"
        name="name"
        type="checkbox"
        value=""
      />
      <label
        for="check-box_1235_name"
      >
        label
      </label>
    </div>
  </div>
</div>
`;

exports[`<CustomRegisterField> > renders and matches snapshot <TextField> 1`] = `
<div>
  <div
    class="_checkbox_531f07"
  >
    <div
      class="_row_531f07"
    >
      <input
        id="check-box_1235_name"
        name="name"
        type="checkbox"
        value=""
      />
      <label
        for="check-box_1235_name"
      >
        label
      </label>
    </div>
  </div>
</div>
`;
